### DISCUSSION TITLE
Nuxt3 handle errros

### DISCUSSION THREAD

**Discussion by artem-kostylev** [nuxt/nuxt#23853] [Questions]
I want to show a fullscreen error (error.vue), how can I do this with useLazyAsyncData ? 

I tried to do both on the screenshot, but it doesn't work 

![image](https://github.com/nuxt/nuxt/assets/26843077/83d5567c-3a08-4d22-85e1-a4b81eacbd31)

I have tried both createError and showError, both inside catch, onResposneError, and in if(error.value) and always works only either on the client or on the server , and I want the same behavior both on the client and on the server

**Comment by artem-kostylev** [comment]
up

**Comment by memic84** [comment]
Have you tried it with added a top-level await for the asyncData?

```
const { data, pending, error, refresh } = await useAsyncData(
  		'mountains',
  		() => $fetch('https://api.nuxtjs.dev/mountains')
)
```

So no await for the $fetch, but on for the asyncData.

**Comment by shunnNet** [comment]
Hi, I did some experiment.

First, add `await`. Then

The thrown `createError` will not show error page when called in `useAsyncData` or `useLazyAsyncData` (both client and server side).
I think the reason may be the error will be handled by `useAsyncData`. The `error` returned by `useAsyncData` will be the thrown `createError`. 

If you want show error page in this situation, use `showError`.

Or throw the returned error

```ts
const { data, pending, error } = await useLazyAsyncData(
  'LazyAsync',
  async () => {
    try {
      throw new Error('Failed')
    } catch (e) {
      console.log('Error occured')
      showError({ statusCode: 404, statusMessage: 'API Failed' })
      
      // ! createError will not work, because it will go to `error`
      // throw createError({ statusCode: 404, statusMessage: 'API Failed', fatal: true })
    }
  },
)

// this line will work like throw createError 
// throw error.value

```

**Reply by artem-kostylev** [reply]
I tried this but it only works on the client render

**Reply by shunnNet** [reply]
Seems like throw normal error will work different from fetching.

I found a bad way to do that:

[sample link](https://stackblitz.com/edit/nuxt-starter-hb8vne?file=pages%2Fsample.vue,app.vue,pages%2Fnormal-error.vue)

```ts
import { $fetch } from "ofetch"
import { useLazyAsyncData, createError } from "#imports"

const fetchPost = () => $fetch('https://jsonplaceholder.typicode.com/posts/500000000000')

const { data, pending, error } = await useLazyAsyncData('post', () => fetchPost())

// work for server 
if (error.value){
  throw createError({ statusCode: 404, statusMessage: "123"})
}

// watch is for client,
// not good solution. for client, maybe show error message directly in template instead
watch(error, (val) => {
  if(val){
      showError({ statusCode: 404, statusMessage: "123"})
  }
})
```

**Reply by artem-kostylev** [reply]
Yes it works, but it's not the best solution...

**Reply by shunnNet** [reply]
If you want same behavior both client and server side, I think you should use `useAsyncData`.
Because the lazy mode will not wait the request at client. That's is how lazy mode works.

**Reply by xjccc** [reply]
a plugins?
```
export default defineNuxtPlugin((nuxtApp) => {
  nuxtApp.vueApp.config.errorHandler = (error, context) => {
    if (error) {
         throw createError({ statusCode: 404, statusMessage: "123"})
     }
  }
})
```

**Comment by artem-kostylev** [comment]
@danielroe can you help with this?

**Reply by SergioWaters** [reply]
Seems like useFetch and useAsyncData are called in beforeCreated hook, while createError in created.
